palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/util/api.py

```diff
@@ -0,0 +1,301 @@
+            ]
+
+        done = False
+        all_incidents = []
+
+        # Keep paging through and requesting incidents until we have requested all incidents which match our query
+        while not done:
+            resp_json = self._post_to_api(endpoint, post_body)
+            if resp_json is not None:
+                total_count = resp_json.get("reply", {}).get("total_count", -1)
+                all_incidents.extend(resp_json.get("reply", {}).get("incidents", []))
+                # If the number of incidents we have received so far is greater than or equal to the total number of
+                # incidents which match the query, then we can stop paging.
+                if len(all_incidents) >= total_count or total_count < 0:
+                    done = True
+
+                # Update the indices of search_from and search_to to we can request the next page
+                search_from = search_from + batch_size
+                search_to = search_to + batch_size if search_to + batch_size < total_count else total_count
+
+                # Add the updated page indices to the request body for when we request the next page
+                post_body["request_data"]["search_from"] = search_from
+                post_body["request_data"]["search_to"] = search_to
+            else:
+                done = True
+
+            # Back-off between making requests to the API.
+            time.sleep(1)
+
+        return all_incidents
+
+    def allow_or_block_file(self, file_hash, comment, incident_id=None, block_file=True):
+        if block_file:
+            endpoint = "/public_api/v1/hash_exceptions/blocklist/"
+        else:
+            endpoint = "/public_api/v1/hash_exceptions/allowlist/"
+
+        post_body = {
+            "request_data": {
+                "hash_list": [file_hash],
+                "comment": comment,
+            }
+        }
+
+        if incident_id:
+            post_body["request_data"]["incident_id"] = incident_id
+
+        block_action = "Block" if block_file else "Allow"
+
+        self.logger.info(f"Taking {block_action} action on file: {file_hash}")
+        self.logger.info(f"Endpoint: {endpoint}")
+
+        result = self._post_to_api(endpoint, post_body)
+
+        if result.get("reply"):  # reply will be true or false
+            self.logger.info(f"{block_action} action was successful")
+            return True
+        self.logger.warning(f"{block_action} action failed")
+        self.logger.warning(f"Result: {result}")
+        return False
+
+    ###########################
+    # Private Methods
+    ###########################
+    def _isolate_multiple_endpoints(self, endpoints, isolation_state):
+        if isolation_state:
+            api_endpoint = "/public_api/v1/endpoints/isolate/"
+        else:
+            api_endpoint = "/public_api/v1/endpoints/unisolate/"
+
+        endpoint_ids = []
+        for ep in endpoints:
+            endpoint_ids.append(ep.get("endpoint_id"))
```

Comment:
```suggestion
        endpoint_ids = [ep.get("endpoint_id") for ep in endpoints if ep.get("endpoint_id") is not None]
```
Shorter and ensure None values don't make their way into the list

Reply:
I disagree with this comment. It's slicker code, but harder to debug. You can't trace through the lambda.